### DISCUSSION TITLE
FoxyTab: what is a "Tab Host"?

### DISCUSSION THREAD

**Discussion by philCryoport** [erosman/support#660] [Q&A] [answered]
Hi all,

When I went to look for Firefox extensions that would prevent me from closing a tab, FoxyTab was at the top of the list.

Does "Pin Tab" prevent one from closing a tab?
(I use Cmd-W on my Mac -- and I'd also like to prevent mouse-clicking close-tab or the tab's X)

What is a "Tab Host"?  The "Help" doesn't explain what that is.  Help?

**Comment by erosman** [comment]
Toolbar popup has the option to include/exclude pinned tabs.

Where did you see "Tab Host"?
It should mean host of the current tab.

**Reply by philCryoport** [reply]
> Toolbar popup has the option to include/exclude pinned tabs.
> 
> Where did you see "Tab Host"? It should mean host of the current tab.

Here's where I saw in FoxyTab the concept of a "Tab Host":
![image](https://github.com/user-attachments/assets/813b0289-a644-415d-acf3-3f8063648f82)

So what exactly is a "Tab Host"?  

Why would I use a "Tab Host"?  

What does "Host Keep" do?

What is the difference between "Move Host to New Window" -- versus the option to "Move to New Window to Left" or "Move to New Window to Right":
![image](https://github.com/user-attachments/assets/508a76e1-9b28-4dfb-a706-e8de44cf1221)

**Comment by erosman** [comment]
Host is the domain of the Tab URL. For example, in `https://www.example.com/blahnlah`, the host is `www.example.com`.

Host operations are as following:

![image](https://github.com/user-attachments/assets/f1e4d18c-d337-4eb1-9e98-fd8b90884d73)


They are used when you want to process tabs based on their HOST, for example close all tabs with  `www.example.com`, or keep  all tabs with`www.example.com` and close all other tabs.

**Reply by philCryoport** [reply]
Ok @erosman  that make sense now.

Host Tab operations won't do what I'm looking to do.  What will?

My needs:
```
* Toggle a Tab as "un-close-able" via any of the following:
  * Clicking the X in the Tab
  * While viewing the Tab, hitting Ctrl-W (Windows & Linux) / Cmd-W (macOS)
  * Exiting the browser completely
...somehow notifying me that the Tab is toggled as "un-close-able" when I try any of the above

...with the ability to toggle that "un-close-able" Tab back to "close-able" (i.e. normal behavior)
```

Is there a FoxyTab feature that I'm just not seeing?  Is this what "Pin Tab" does?

**Comment by erosman** [comment, marked as answer]
At the moment, there is no such feature in FoxyTab.

"Pin Tab" is similar to the built-in Firefox "Pin Tab".
